woocommerce/woocommerce-ios
# Product Subscriptions: Add trial info view for subscription product

Closes: #11152 


## Description

- Adds a new row in the Product detail screen to display a subscription product's free trial info. 
- Adds a new view to view/edit the trial info. 

**Changes**

- Adds a new SwiftUI view and view model for free trial view. `SubscriptionTrialView`
- Edit product form related files to show the free trial row in the product form.
- Add/update unit tests.

## Testing instructions
- Enable `subscriptionProducts` feature flag by returning `true` from [this line](https://github.com/woocommerce/woocommerce-ios/blob/trunk/Experiments/Experiments/DefaultFeatureFlagService.swift#L90) 
- Launch the app and login 
- Navigate to products
- Tap on a simple subscription product (Create one from `wpadmin` if needed)
- Check that the Free trial info is displayed in the product details screen
- Tap on the Free trial row
- Edit the Free trial info, tap `Done`, and Save the product
- Validate that the product is updated with the new free trial info (in the wpadmin page)
- Open the product again and try to enter an invalid Free trial period. 
```
The trial period can not exceed: 90 days, 52 weeks, 24 months or 5 years.
```
- Ensure that you see an error message row and the `Done` button is disabled until you enter a valid free trial value. 

## Screenshots


| Screen | Dark | Light |
|--------|--------|--------|
| Product form | ![Simulator Screenshot - iPhone 14 Pro - 2023-11-16 at 09 14 12](https://github.com/woocommerce/woocommerce-ios/assets/524475/6148f0b7-3130-44ae-bbcc-2c0adaed31c1) | ![Simulator Screenshot - iPhone 14 Pro - 2023-11-16 at 09 14 18](https://github.com/woocommerce/woocommerce-ios/assets/524475/512353dd-e49f-45a3-bd6c-9c6e4f2e877f) |
| Free trial view | ![Simulator Screenshot - iPhone 14 Pro - 2023-11-16 at 09 14 41](https://github.com/woocommerce/woocommerce-ios/assets/524475/86e8be6a-45a2-43c0-8d90-172705955872) | ![Simulator Screenshot - iPhone 14 Pro - 2023-11-16 at 09 14 35](https://github.com/woocommerce/woocommerce-ios/assets/524475/8620f561-8d57-43b5-bbda-45be7efece06) |

https://github.com/woocommerce/woocommerce-ios/assets/524475/dd90b959-4fb1-4306-8194-3d75fcfa8c92



---
